main.js

```diff
@@ -796,10 +796,20 @@ addStealExtension(function applyTraceExtension(loader) {
 // Steal JSON Format
 // Provides the JSON module format definition.
 addStealExtension(function (loader) {
-  var jsonTest = /^[\s\n\r]*[\{\[]/;
   var jsonExt = /\.json$/i;
   var jsExt = /\.js$/i;
 
+  // taken from prototypejs
+  // https://github.com/sstephenson/prototype/blob/master/src/prototype/lang/string.js#L682-L706
+  function isJSON(str) {
+    if (!str) return false;
+
+    str = str.replace(/\\(?:["\\\/bfnrt]|u[0-9a-fA-F]{4})/g, '@');
```

Comment:
this might be dated information... but it used to be true that defining a regex inside a function caused it to be created each time. I'm not sure if they are treated as constants by modern js engines or not (someone to look into perhaps).  I usually write in this form for this kind of thing:

```js
var isJSON = (function(){
  var firstRegExp = /.../;

  return function(str){
  };
})();
```